lib/index.js

```diff
@@ -377,11 +425,37 @@ MongoStorage.prototype.download.accepts = [
   }
 ];
 
+
+
 MongoStorage.prototype.download.http = {
   verb: 'get',
   path: '/:container/download/:file'
 };
 
+//New API endpoint to download file by id
+MongoStorage.prototype.downloadFileById.shared = true;
+
+MongoStorage.prototype.downloadFileById.accepts = [
+  {
+    arg: 'container',
+    type: 'string'
+  }, {
+    arg: 'fileId',
+    type: 'string'
+  }, {
+    arg: 'res',
+    type: 'object',
+    http: {
+      source: 'res'
+    }
+  }
+];
+
+MongoStorage.prototype.downloadFileById.http = {
+  verb: 'get',
+  path: '/:container/downloadFile/:fileId'
```

Comment:
oops. The path is supposed to be /:container/downloadFileById/:fileId. I will fix and resend the pull request.